http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -150,6 +169,13 @@ public void apply(Request request)
         };
     }
 
+    private static boolean networkIssue(Throwable throwable)
+    {
+        requireNonNull(throwable, "throwable is null");
+        return Throwables.getCausalChain(throwable).stream()
+                .anyMatch(ConnectException.class::isInstance);
```

Comment:
We might want to use `SocketException` here (of which `ConnectException` is a subclass)